revert gentoo mupdf-1.18-Makefile.patch
diff --git a/Makefile b/Makefile
index 174ab4d..a45b8c4 100644
--- a/Makefile
+++ b/Makefile
@@ -3,7 +3,7 @@
 -include user.make
 
 ifndef build
-  build := debug
+  build := release
 endif
 
 default: all
@@ -265,19 +265,16 @@ $(THIRD_GLUT_LIB) : $(THIRD_GLUT_OBJ)
 $(THREAD_LIB) : $(THREAD_OBJ)
 $(PKCS7_LIB) : $(PKCS7_OBJ)
 else
-MUPDF_LIB = libmupdf.so.${GENTOO_PV}
-MUPDF_STATIC = $(OUT)/libmupdf.a
-THIRD_LIB = 
+MUPDF_LIB = $(OUT)/libmupdf.a
 LIBS_TO_INSTALL_IN_LIB = $(MUPDF_LIB) $(THIRD_LIB)
+THIRD_LIB = $(OUT)/libmupdf-third.a
 ifneq ($(USE_SYSTEM_GLUT),yes)
 THIRD_GLUT_LIB = $(OUT)/libmupdf-glut.a
 endif
 THREAD_LIB = $(OUT)/libmupdf-threads.a
 PKCS7_LIB = $(OUT)/libmupdf-pkcs7.a
 
-$(MUPDF_LIB) : $(MUPDF_OBJ) $(THIRD_OBJ)
-	$(QUIET_LINK) $(CC) $(LDFLAGS) --shared -Wl,-soname -Wl,$(MUPDF_LIB) -o $@ $^ $(THIRD_LIBS) $(LIBS)
-$(MUPDF_STATIC): $(MUPDF_OBJ) $(THIRD_OBJ)
+$(MUPDF_LIB) : $(MUPDF_OBJ)
 $(THIRD_LIB) : $(THIRD_OBJ)
 $(THIRD_GLUT_LIB) : $(THIRD_GLUT_OBJ)
 $(THREAD_LIB) : $(THREAD_OBJ)
@@ -434,7 +431,7 @@ install-docs:
 
 	install -d $(DESTDIR)$(docdir)
 	install -d $(DESTDIR)$(docdir)/examples
-	install -m 644 README CHANGES $(DESTDIR)$(docdir)
+	install -m 644 README COPYING CHANGES $(DESTDIR)$(docdir)
 	install -m 644 docs/*.html docs/*.css docs/*.png $(DESTDIR)$(docdir)
 	install -m 644 docs/examples/* $(DESTDIR)$(docdir)/examples
 
