--- bcc-0.25.0-orig/tools/ttysnoop.py	2022-08-11 00:39:43.000000000 +0530
+++ bcc-0.25.0/tools/ttysnoop.py	2022-12-02 16:09:43.000000000 +0530
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3.9
 # @lint-avoid-python-3-compatibility-imports
 #
 # ttysnoop   Watch live output from a tty or pts device.
@@ -120,7 +120,7 @@
  * commit 9bb48c82aced (v5.11-rc4) tty: implement write_iter
  * changed arguments of tty_write function
  */
-#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 11, 0)
+#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 10, 88) //;madhu 220501 was 5.11.0
 int kprobe__tty_write(struct pt_regs *ctx, struct file *file,
     const char __user *buf, size_t count)
 {
