--- gtk+-3.24.38-orig/meson.build	2023-05-22 07:52:08.000000000 +0530
+++ gtk+-3.24.38/meson.build	2023-09-03 00:24:14.661084083 +0530
@@ -832,7 +832,7 @@
 
 profiler_enabled = get_option('profiler')
 if profiler_enabled
-  profiler_dep = dependency('sysprof-capture-3', static: true, required: true)
+  profiler_dep = dependency('sysprof-capture-3', static: false, required: true)
   if profiler_dep.found()
     cdata.set('HAVE_SYSPROF_CAPTURE', profiler_dep.found())
   else
